io-engine/src/lvs/lvs_lvol.rs

```diff
@@ -875,4 +875,36 @@ impl SnapshotOps for Lvol {
     async fn snapshot_iter(self) -> LvolSnapshotIter {
         LvolSnapshotIter::new(self)
     }
+    /// Prepare Snapshot Config for Block/Nvmf Device, before snapshot create.
+    async fn prepare_snap_config(
+        &self,
+        snap_name: &str,
+        txn_id: &str,
+    ) -> Option<SnapshotParams> {
+        // Entity Id will be same as lvol uuid for the replica snapshot.
+        let entity_id = Some(self.uuid());
+        // Current Lvol uuid is the parent for the snapshot.
+        let parent_id = Some(self.uuid());
+
+        let bdev_handle = device_open(self.as_bdev().name(), false)
+            .unwrap()
+            .into_handle()
+            .unwrap();
+        match bdev_handle.flush_io().await {
```

Comment:
Could we put I/O flush inside snapshot operation to make sure this is always done as part of the snapshot creation operation itself ? In such a case this also saves us one extra device open operation.

Reply:
> Could we put I/O flush inside snapshot operation to make sure this is always done as part of the snapshot creation operation itself ? In such a case this also saves us one extra device open operation.

Accepted and Done the modification.